aws/data_source_aws_ec2_instance_spot_price.go

```diff
@@ -0,0 +1,106 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsEc2InstanceSpotPrice() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEc2InstanceSpotPriceRead,
+
+		Schema: map[string]*schema.Schema{
+			"filter": dataSourceFiltersSchema(),
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Since this can also be provided via filter, this should be optional. 👍 

```suggestion
				Optional: true,
```

Reply:
Done :+1: